### PR TITLE
Add generateOwnerToken functions

### LINKED ISSUE #5: Accept organization name and look up installation id
https://docs.github.com/en/rest/apps/apps?apiVersion=2022-11-28#get-an-organization-installation-for-the-authenticated-ap

There seems to be a few different ways to look up an installation, depending on if its installed on an organization, a user, or a repository thereof. Should we support them all?

### PR DESCRIPTION
**Pull request by pbrisbin**
Pre-factors included, the main commit is: 4805cf0b74114c9748db01f4f5a06d5f33da88c2.

### CODE REVIEW

File: src/GitHub/App/Token/Generate.hs [hunk 1]
```diff
@@ -88,36 +109,97 @@ generateInstallationTokenScoped
...
+generateOwnerTokenScoped
+  :: MonadIO m
+  => CreateAccessToken
+  -> AppCredentials
+  -> Owner
+  -> m AccessToken
+generateOwnerTokenScoped create creds owner = do
+  installation <- getInstallation creds $ case (create.repositories, owner) of
+    (repo : _, Org org) -> "/repos/" <> org <> "/" <> repo
+    (repo : _, User username) -> "/repos/" <> username <> "/" <> repo
```
Comment (mjgpy3) [inline src/GitHub/App/Token/Generate.hs:134]: Is it possible that these `: _`s could be throwing away repos that consumers might expect to have been included?

If not, can the types be re-worked to make those states impossible (or at least not appear in the `case` statement)?

If so, is that okay? Could it cause headache?

**Review by mjgpy3 — APPROVED** [review]
Looks good. I left one question, non-blocking, take it or leave it.

File: src/GitHub/App/Token/Generate.hs (hunk 1 above)
Comment (pbrisbin) [reply src/GitHub/App/Token/Generate.hs:134]: It's a little confusing, but I'm mimicking the behavior of `actions/create-github-app-token`, which is:

If `repositories` is present as a scope, use `repositories.first` to get the installation id -- but we are still including all of them when it is used as a scope.

So the `_` are discarding other repositories here (and we're ignoring `repository_ids` entirely), but that all still gets used when getting the actual access-token.

Comment (pbrisbin) [reply src/GitHub/App/Token/Generate.hs:134]: This logic has a comment in the action, albeit one that that describes the _what_ more than the _why_. I think I'll add a (better) comment here.

Comment (mjgpy3) [reply src/GitHub/App/Token/Generate.hs:134]: Makes sense, thanks Pat.

**Pull request merged**